File: src/Services/Portal/Portal.Domain/Enums/ERoleType.cs
using System.ComponentModel;

namespace Portal.Domain.Enums
{
    public enum ERoleType
    {
        [Description("User")]
        User = 0,
        [Description("Partner")]
        Partner = 1,
        [Description("Admin")]
        Administrator = 2,
        [Description("User Premium")]
        UserPremium = 3,
        [Description("User Super Premium")]
        UserSuperPremium = 4
    }
}

File: app/models/enums/ERoleType.ts
import { EnumDictionary } from "./EnumDictionary";

export enum ERoleType {
  NoneRole = -1,
  User = 0,
  UserPremium = 1,
  UserSuperPremium = 2,
  Partner = 3,
  Administrator = 99
}

export const roleTypeEnumMapping: EnumDictionary<ERoleType, string> = {
  [ERoleType.NoneRole]: "",
  [ERoleType.User]: "User",
  [ERoleType.UserPremium]: "User Premium",
  [ERoleType.UserSuperPremium]: "User Super Premium",
  [ERoleType.Partner]: "Partner",
  [ERoleType.Administrator]: "Administrator"
}